libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -828,6 +828,25 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
     }
 
+    fun previewNavigationRoutes(
+        routes: List<NavigationRoute>,
+        initialLegIndex: Int = 0,
+    ) {
+        //TODO: parse alternatives metadata using NN and set it route alternatives controller
+        directionsSession.setRoutes(routes, initialLegIndex, RoutesExtra.ROUTES_UPDATE_REASON_NEW)
```

Comment:
Which NavigationSessionState are we gonna be in after this invocation? The prev one, right, nothing changes here?

Reply:
That's a very good question. I would vote for introducing a new state, but it requires deprecation of old states and introducing a new one.